internal/pkg/describe/project.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package describe
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"text/tabwriter"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
+)
+
+type Environment struct {
+	Name      string `json:"name"`
+	AccountID string `json:"accountID"`
+	Region    string `json:"region"`
+}
+
+type Application struct {
+	Name string `json:"name"`
+	Type string `json:"type"`
+}
+
+// Project contains serialized parameters for a project.
+type Project struct {
+	Name string         `json:"name"`
+	URI  string         `json:"uri"`
+	Envs []*Environment `json:"environments"`
+	Apps []*Application `json:"applications"`
+}
+
+// JSONString returns the stringified Project struct with json format.
+func (p *Project) JSONString() (string, error) {
+	var envsToSerialize []*Environment
+	for _, env := range p.Envs {
+		envsToSerialize = append(envsToSerialize, &Environment{
+			Name:      env.Name,
+			AccountID: env.AccountID,
+			Region:    env.Region,
+		})
+	}
+	var appsToSerialize []*Application
+	for _, app := range p.Apps {
+		appsToSerialize = append(appsToSerialize, &Application{
+			Name: app.Name,
+			Type: app.Type,
+		})
+	}
+	projectToSerialize := &Project{
+		Name: p.Name,
+		URI:  p.URI,
+		Envs: envsToSerialize,
+		Apps: appsToSerialize,
+	}
+	b, err := json.Marshal(projectToSerialize)
+	if err != nil {
+		return "", fmt.Errorf("marshal project: %w", err)
+	}
+	return fmt.Sprintf("%s\n", b), nil
```

Comment:
What's the reason for all the mapping in here rather than a simple `json.Marshal(p)` call?